int32 id
float64 x
float64 y
float64 yaw
---
float64[] X
float64[] P
